src/com/goide/inspections/GoCommentStartInspection.java

```diff
@@ -55,35 +55,80 @@ public void visitPackageClause(@NotNull GoPackageClause o) {
 
       @Override
       public void visitCompositeElement(@NotNull GoCompositeElement o) {
-        if (!(o instanceof GoNamedElement) || !((GoNamedElement)o).isPublic()) {
-          return;
+        super.visitCompositeElement(o);
+        if (!shouldInspect(o)) return;
+
+        boolean isBlockComment = false;
+        int elementsInBlock = 0;
+        PsiElement specElement = o;
+        if (o instanceof GoVarDefinition || o instanceof GoConstDefinition) {
```

Comment:
For GoTypeDeclaration we retrieve the comments correctly, whereas for those two, unless we do that change, can't have this handled correctly:
```go
var (
	A int
	// demo comment -> this comment is not correctly handled
	C int
)

type (
	Deeeeo struct {
	}
	// demo stuff -> this comment is handled correctly
	Ehlo struct{
	}
)
```